chat-plugins/uno.js

```diff
@@ -0,0 +1,694 @@
+	"R4",
+	"R5",
+	"R6",
+	"R7",
+	"R8",
+	"R9",
+	"RS",
+	"RR",
+	"R+2",
+	"Z+4",
+	"Y1",
+	"Y2",
+	"Y3",
+	"Y4",
+	"Y5",
+	"Y6",
+	"Y7",
+	"Y8",
+	"Y9",
+	"YS",
+	"YR",
+	"Y+2",
+	"Z+4",
+	"G1",
+	"G2",
+	"G3",
+	"G4",
+	"G5",
+	"G6",
+	"G7",
+	"G8",
+	"G9",
+	"GS",
+	"GR",
+	"G+2",
+	"Z+4",
+	"B1",
+	"B2",
+	"B3",
+	"B4",
+	"B5",
+	"B6",
+	"B7",
+	"B8",
+	"B9",
+	"BS",
+	"BR",
+	"B+2",
+	"Z+4",
+	"R1",
+	"R2",
+	"R3",
+	"R4",
+	"R5",
+	"R6",
+	"R7",
+	"R8",
+	"R9",
+	"RS",
+	"RR",
+	"R+2",
+	"ZW",
+	"R0",
+	"Y1",
+	"Y2",
+	"Y3",
+	"Y4",
+	"Y5",
+	"Y6",
+	"Y7",
+	"Y8",
+	"Y9",
+	"YS",
+	"YR",
+	"Y+2",
+	"ZW",
+	"Y0",
+	"G1",
+	"G2",
+	"G3",
+	"G4",
+	"G5",
+	"G6",
+	"G7",
+	"G8",
+	"G9",
+	"GS",
+	"GR",
+	"G+2",
+	"ZW",
+	"G0",
+	"B1",
+	"B2",
+	"B3",
+	"B4",
+	"B5",
+	"B6",
+	"B7",
+	"B8",
+	"B9",
+	"BS",
+	"BR",
+	"B+2",
+	"ZW",
+	"B0",
+];
+//declare html for DRAW and PASS button.
+const drawButton = '<center><button style="background: black; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 56px ; border-radius: 5px , auto" name="send" value="/uno draw"><font color="white">Draw</font></button></center>';
+const passButton = '<center><button style="background: red; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 56px ; border-radius: 5px , auto" name="send" value="/uno pass"><font color="white">PASS</font></button></center>';
+
+//get colour for html'd text and backgrounds
+function getColour(colour, background) {
+	if (!background && colour === "Y") return "orange";
+	let colourTable = {
+		"R": "red",
+		"Y": "yellow",
+		"B": "blue",
+		"G": "green",
+		"Z": "black",
+	};
+	return colourTable[colour];
+}
+
+//build html button for the card
+function buildCard(id, notClickable) {
+	let colour = getColour(id.charAt(0), true);
+	let value = id.slice(1);
+	let fontColour = colour === "yellow" ? "black" : "white";
+	let command = "/uno play " + id;
+	let clickable = notClickable ? ">" : "name=\"send\" value=\"" + command + "\">";
+	let title = "title=\"" + getCardName(id) + "\"";
+	let buttonFace = value === "W" ? '<img src="https://www.script-tutorials.com/demos/317/thumb.png" height="27" width="27">' : '<font color="' + fontColour + "\" size=6>" + value + "</font>";
+	let button = "<button " + title + " style=\"background: " + colour + "; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; height: 80px ; width: 56px ; border-radius: 5px , auto\"" + clickable + buttonFace + "</button>";
+	return button;
+}
+
+//build an display of multiple cards
+function buildHand(array, notClickable) {
+	let hand = [];
+	array.sort().forEach(function (c) {
+		hand.push(buildCard(c, notClickable));
+	});
+	return hand.join("&nbsp;");
+}
+
+
+function initDeck(playerCount) {
+	playerCount = Math.ceil(playerCount / 7);
+	let tDeck = [];
+	for (let i = 0; i < playerCount; i++) {
+		tDeck = tDeck.concat(deck);
+	}
+	return tDeck;
+}
+
+function shuffleDeck(dArray) {
+	let tDeck = [];
+	let reiterations = dArray.length;
+	for (let i = 0; i < reiterations; i++) {
```

Comment:
There's no reason to reimplement Fisher-Yates if we already depend on a library that implements it (namely, `sugar`).